inspirehep/modules/records/serializers/schemas/json.py

```diff
@@ -76,7 +76,7 @@ def processed_refs(self, data):
             data['metadata']['references'] = []
             return data
 
-        reference_records = self.resolve_records(references)
+        reference_records = self.resolve_records(data)
```

Comment:
hmm should be ``data['metadata']``, no? cause you have [this](https://github.com/inspirehep/inspire-next/pull/3509/files#diff-65ba6e9a03cccd086772d35d5fd5ba60R119):

```python
resolved_records = get_linked_records_in_field(data, 'references.record')
```

Reply:
it passes all tests though, so I guess this is not really tested.

Reply:
I changed the unit test mocking the `get_linked_records_in_field` function into an integration test without mocks. With @turtle321's `TestRecordMetadata` factory, it's almost easier to create test data in the db than mocking functions reading from the db.